### PR TITLE
@class is not associated with the class, and is always nil. Get rid of it

### PR DESCRIPTION
**Pull request by vipulnsward**

### CODE REVIEW

**Comment by yahonda** [comment]
Since Arel 7.1.0 has been released, it gets this warning many times.

```ruby
/home/yahonda/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/arel-7.1.0/lib/arel/nodes/casted.rb:14: warning: instance variable @class not initialized
```

File: lib/arel/nodes/casted.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def initialize val, attribute
       def nil?; @val.nil?; end
 
       def hash
-         [@class, @val, @attribute].hash
+         [@val, @attribute].hash
```
Comment (rafaelfranca) [inline lib/arel/nodes/casted.rb]: We should change it to `self.class` like binary https://github.com/vipulnsward/arel/blob/171b422386f0e7183993ab554b152b7f310eb8df/lib/arel/nodes/binary.rb

Comment (rafaelfranca) [inline lib/arel/nodes/casted.rb]: Well, we don't. Only if we had a subclass. I'll merge.

**Comment by rafaelfranca** [comment]
@matthewd already fixed it.

**Comment by pas256** [comment]
Fantastic, thank you! I am really looking forward to a new release of this gem.

**Comment by rafaelfranca** [comment]
I just pushed 7.1.1

**Pull request closed without merging**